src/main/java/com/nvidia/spark/rapids/jni/nvml/NVML.java

```diff
@@ -58,25 +55,36 @@ public class NVML {
     private static boolean nativeLibraryLoaded = false;
 
     /**
-     * Initialize NVML library
+     * Initialize NVML library and load native dependencies
      */
     public static synchronized boolean initialize() {
```

Comment:
I don't think I've quite gotten the failure case, when the native library is not available.  Where `NVML::initialize()` is called from, in `spark-rapids`, currently?

In the event that the library isn't loaded, `initialize()` returns `false`, and none of the other methods will/should be called.  Is that right?